app/util/decorators.py

```diff
@@ -27,6 +27,47 @@ def function_with_retries(*args, **kwargs):
     return method_decorator
 
 
+def retry_on_exception_exponential_backoff(exceptions, initial_delay=0.1, max_delay=10):
```

Comment:
Yeah, I was worried about adding too many arguments, but four isn't too bad, and it's likely not going to increase in the future.